src/main/java/seedu/task/logic/commands/EditTaskCommand.java

```diff
@@ -98,19 +98,26 @@ private Task editTask(ReadOnlyTask targetTask) {
         if (!isNameToBeEdit) {
             newName = targetTask.getTask();
         }
-        if (!isDescriptionToBeEdit) {
-            newDescription = targetTask.getDescription();
+        if (!isDescriptionToBeEdit && targetTask.getDescription().isPresent()) {
```

Comment:
Have an idea, why not passing an empty string "" into the Task constructor directly? And the constructor will initialise the Optional<Description> properly. 
Therefore you do not have to call multiple constructors. 
Does this make sense?